File: codes/sates_test_cs/testcode_list.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace sates.test.cs
{
    /** \addtogroup CODE
    *  @{
    */
    /** \addtogroup sates
    *  @{
    */
    /** \addtogroup test
    *  @{
    */
    /** \addtogroup cs
    *  @{
    */

    /**
    @revision
    |  날짜  | 내용  | 담당자   | 검수자  | 
    |------------|------------|------------|------------|
    |2018년 7월 30일 | run 함수 내, 테스트 결과 저장 루틴 추가 | 김석환  |  사용자  |
     */

    /// <summary>
    /// TCP / IP 소켓을 통해 json 커맨드 등을 수신
    /// </summary>
    public class testcode_list
    {
        static List<testcode> testcode_list_var = new List<testcode>();
        public delegate void basic_func_t();

        public static basic_func_t global_init_func = null;
        public static basic_func_t global_terminate_func = null;

        public static sates.test.cs.report.reporter reporter {get; set;}

        public static void create()
        {
            global_init_func?.Invoke();
        }

        public static void destroy()
        {
            global_terminate_func?.Invoke();
        }

        public static void add_testcode(testcode code)
        {
            testcode_list_var.Add(code);
        }

        public static testcode get_testcode(string test_case_name)
        {
            testcode retval = null;
            foreach (var item in testcode_list_var)
            {
                if(item.test_case_name == test_case_name)
                {
                    retval = item;
                    break;
                }
            }
            return retval;
        }

        public static void run()
        {
            
            foreach (var item in testcode_list_var)
            {
                SATES.CUR_ITEM = item;
                SATES.RESULT = TEST_RESULT.OK;

                Console.WriteLine("");
                Console.WriteLine("======================================================================");
                Console.WriteLine("TEST BEGIN : " + item.GetType().ToString());

                item.init();
                item.run();
                item.terminate();

                item.result = SATES.RESULT;

                Console.WriteLine("TEST END   : " + item.GetType().ToString());
                Console.WriteLine("======================================================================");
                Console.WriteLine("");

                sates.test.cs.test_result_reporter_josn.add_result(item);
            }
            string report_str = sates.test.cs.test_result_reporter_josn.get_report_string();
            reporter.report(report_str);
        }
    }
    /** @} */
    /** @} */
    /** @} */
    /** @} */
}

File: codes/sates_test_java/src/com/sates/test/java/testcode_list.java
package com.sates.test.java;

import java.util.LinkedList;
import java.util.List;

public class testcode_list
{
    private static List<testcode> testcode_list_var = new LinkedList<>();
    public static basic_func_interface init_func;
    public static basic_func_interface terminate_func;


    public static void create()
    {
        init_func.call();
    }

    public static void destroy()
    {
        terminate_func.call();
    }

    public static void add_testcode(testcode code)
    {
        testcode_list_var.add(code);
    }

    public static testcode get_testcode(String testcase_name)
    {
        testcode retval = null;
        for (testcode code : testcode_list_var)
        {
            if(code.test_case_name == testcase_name)
            {
                retval = code;
                break;
            }
        }
        return retval;
    }

    public static void run()
    {
        for (testcode code : testcode_list_var)
        {
            SATES.CUR_ITEM = code;
            SATES.RESULT = TEST_RESULT.OK;

            System.out.println("");
            System.out.println("======================================================================");
            System.out.println("TEST BEGIN : " + code.getClass().getName());

            code.init();
            code.run();
            code.terminate();

            code.result = SATES.RESULT;

            System.out.println("TEST END   : " + code.getClass().getName());
            System.out.println("======================================================================");
            System.out.println("");

            test_result_reporter_json.report(code);
        }
    }
}
